### PR TITLE
Changes needed to be deployable on Elastic Beanstalk

### PR DESCRIPTION
**Pull request by kompfner**
The site is now available at [demos.daily.co](https://demos.daily.co). The main remaining task is to configure CDN caching of static files.

### CODE REVIEW

File: static-demos/package.json [hunk 1]
```diff
@@ -5,8 +5,10 @@
     "dev": "micro-dev -p 3001"
   },
   "devDependencies": {
```
Comment (kwindla) [inline static-demos/package.json]: Just a quick comment that using micro-dev for the original `daily-js/demos` directory was an experiment, and I don't know that there turned out to be any advantage over using express. I'm fine if we want to standardize on using express everywhere.

**Comment by kwindla** [comment]
Regarding cdn caching of static assets ... coincidentally, last week I started setting up a Cloudfront distribution to cache our production static assets.

  https://console.aws.amazon.com/cloudfront/home?#distribution-settings:E32ENEX1U742P3

The main motivation is that we've gotten reports from some international users that call-machine-bundle.js loads very slowly -- sometimes so slowly that they can't ever get the complete bundle before the request times out.

But I got stuck trying to figure out how to serve a different bundle for staging and production Elastic Beanstalk 

https://c.daily.co/static is set up to serve from cloudfront. For example:

  https://c.daily.co/static/call-machine-bundle.js

There's clearly a win, here. On my home network, download of the big call-machine-bundle.js file from our Elastic Beanstalk servers averages ~750ms. Download from Cloudfront averages ~250ms.

Of course, the other important thing to do is to reduce the size of our js bundles!

File: static-demos/package.json (hunk 1 above)
Comment (kompfner) [reply static-demos/package.json]: Totally. I understand why the choice was made—seems like the easiest way to get a simple but workable "index" site of all files in a directory. I mostly wanted to port over the work as-is with the intention later taking another pass and surfacing each demo in the site's new top-level index page.

**Comment by kompfner** [comment]
Re: the CDN issue you raised...if I understand correctly, we essentially want our CDN distribution to be able to serve JS from different origins based on the request's subdomain, right? Like `foo.staging.daily.co/static/call-machine-bundle.js` should serve bundle x and `foo.daily.co/static/call-machine-bundle.js` should serve bundle y?

If so, then based on a little research it seems like we may be forced to create separate distributions for each environment. The only customizability we seem to have over cache behavior is based on "path pattern", but that seems to only seems to match path and not the rest of the url (https://stackoverflow.com/a/29503816).

Is `c.daily.co` the subdomain you're using to test out CDN caching before applying to live traffic?

**Comment by kwindla** [comment]
cache control change looks good to me.

**Comment by kwindla** [comment]
Regarding cdn ...

Yes, `c.daily.co` is a subdomain I pointed at the Cloudfront distribution for testing! I was thinking that we might serve from that domain, too, for production caching, but I'm not sure.

Here are the goals and constraints I was juggling, thinking about how to move forward with cdn caching of productions static assets:

- by far, the most important thing to cache/serve from the cdn are the big js bundles
- we have a complicated `app-index.js` file that defines a large number of express routes and rules ... it would be nice to change that as little as possible
- I don't want to change our DNS to point *.daily.co at Cloudfront, and let Cloudfront proxy our dynamic requests, because every time I've done anything complicated with a cdn, something non-obvious has broken. We might eventually do this, but only after a moderate amount of planning and testing!

`app-index.js` serves: old `meet.pluot.co` resources for backwards compatibility with our very old meeting links; `tv.daily.co` resources used by our hardware; internal API calls from `*.pluot.blue`; external api calls from `api.daily.co`; signaling via `/_ks_ and /_ks_beacon`; `dashboard.daily.co`; `*.daily.co` meetings on customer "custom domains", and maybe others I'm forgetting.

From the perspective of not changing `app-index.js` right now, the easiest thing might be to serve any static resources we want to serve from a cdn from `c.daily.co/...`

I *think* the best way to do that is to change our webpack config so we are injecting a <script> tag into our little html root pages with a little bit of javascript. Injecting the <script> tag programatically will let us point the script src at `c.daily.co` if we're on prod, and keep the relative url we have now for everything that's not prod. 

Longer term, we'll want to:
- end of life `pluot.co` meeting links and all tv (our hardware) functionality
- refactor our code to split apart call-machine and dashboard into different "servers" (I think).

**Pull request merged**